ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -163,7 +163,8 @@ type RayClusterReconcilerOptions struct {
 
 // Reconcile used to bridge the desired state with the current state
 func (r *RayClusterReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	ctx = ctrl.LoggerInto(ctx, r.Log) // TODO: add request namespace here
+	logger := r.Log.WithValues("RayCluster", request.NamespacedName)
```

Comment:
I use `"RayCluster"` instead of `"raycluster"` because controller-runtime also uses `"RayCluster"` if the function `Reconcile` returns both a non-zero result and a non-nil error.

```
{"level":"info","ts":"2024-02-27T06:34:34.039Z","msg":"Warning: Reconciler returned both a non-zero result and a non-nil error. The result will always be ignored if the error is non-nil and the non-nil error causes reqeueuing with exponential backoff. For more details, see: https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler","controller":"raycluster","controllerGroup":"ray.io","controllerKind":"RayCluster","RayCluster":{"name":"raycluster-complete","namespace":"default"},"namespace":"default","name":"raycluster-complete","reconcileID":"c86f5665-53fe-4aca-bff6-aa91b43c0d0c"}
{"level":"error","ts":"2024-02-27T06:34:34.039Z","msg":"Reconciler error","controller":"raycluster","controllerGroup":"ray.io","controllerKind":"RayCluster","RayCluster":{"name":"raycluster-complete","namespace":"default"},"namespace":"default","name":"raycluster-complete","reconcileID":"c86f5665-53fe-4aca-bff6-aa91b43c0d0c","error":"Operation cannot be fulfilled on rayclusters.ray.io \"raycluster-complete\": the object has been modified; please apply your changes to the latest version and try again","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:329\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227"}
```

Reply:
There's a lot of duplicated metadata here. `RayCluster` contains namespace and name`, but there's also `namespace` and `name` as well.

Reply:
The two log messages originate from the default logger of controller-runtime. Hence, we will only see them when `Reconcile` returns a non-nil error. KubeRay logs will look like this https://gist.github.com/kevin85421/ec7bcdfdd2f67bcc3e1769d2aee3bd57 (no duplicate metadata).

Reply:
gotcha, thanks for clarifyign!